Travis failed.
e.g. `ROS_DISTRO=indigo ROSWS=wstool BUILDER=catkin USE_DEB=true ROS_REPOSITORY_PATH=http://packages.ros.org/ros/ubuntu`:
```
++ rosdep install -r -n --from-paths src --ignore-src --rosdistro indigo -y
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
hironx_ros_bridge: Cannot locate rosdep definition for [tf]
hironx_calibration: Cannot locate rosdep definition for [orocos_kdl]
hironx_moveit_config: Cannot locate rosdep definition for [rostest]
rtmros_hironx: Cannot locate rosdep definition for [catkin]
```
```
++ catkin_make
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
CMake Error: your CXX compiler: "CMAKE_CXX_COMPILER-NOTFOUND" was not found.   Please set CMAKE_CXX_COMPILER to a valid compiler path or name.
```
```
-- Could not find the required component 'hrpsys_ros_bridge'. The following CMake error indicates that you either need to install the package with the same name or change your environment so that it can be found.
CMake Error at /opt/ros/indigo/share/catkin/cmake/catkinConfig.cmake:83 (find_package):
  Could not find a package configuration file provided by "hrpsys_ros_bridge"
  with any of the following names:

    hrpsys_ros_bridgeConfig.cmake
    hrpsys_ros_bridge-config.cmake

  Add the installation prefix of "hrpsys_ros_bridge" to CMAKE_PREFIX_PATH or
  set "hrpsys_ros_bridge_DIR" to a directory containing one of the above
  files.  If "hrpsys_ros_bridge" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  rtmros_hironx/hironx_ros_bridge/CMakeLists.txt:4 (find_package)
```

--- a/hironx_ros_bridge/CMakeLists.txt
+++ b/hironx_ros_bridge/CMakeLists.txt
@@ -44,7 +44,8 @@ configure_file(conf/xml.in                      ${PROJECT_SOURCE_DIR}/conf/${ROB
 configure_file(conf/nosim.xml.in                ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.xml)
 configure_file(conf/conf.in                     ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.conf)
 configure_file(conf/nosim.conf.in               ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.conf)
-add_custom_target(${PROJECT_NAME}_model_files ALL DEPENDS ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.RobotHardware.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.RobotHardware.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.xml ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.xml ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.conf)
+configure_file(conf/rtc_on_off_test.conf.in     ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_rtc_on_off_test.conf)
+add_custom_target(${PROJECT_NAME}_model_files ALL DEPENDS ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.RobotHardware.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.RobotHardware.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.xml ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.xml ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_nosim.conf ${PROJECT_SOURCE_DIR}/conf/${ROBOT_NAME}_rtc_on_off_test.conf)
 
 include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 
@@ -104,6 +105,7 @@ add_rostest(test/test-hironx-ros-bridge.test)
 add_rostest(test/test-hironx-ros-bridge-send.test)
 add_rostest(test/test-hironx-ros-bridge-pose.test)
 add_rostest(test/test-hironx-ros-bridge-controller.test)
+add_rostest(test/test-hironx-rtc-on-off.test)
 
 if (CATKIN_ENABLE_TESTING)
   catkin_add_nosetests(test/test_no_ros.py)

--- a/hironx_ros_bridge/conf/rtc_on_off_test.conf.in
+++ b/hironx_ros_bridge/conf/rtc_on_off_test.conf.in
@@ -1,0 +1,10 @@
+model: file://@PROJECT_SOURCE_DIR@/models/@ROBOT_NAME@.dae
+dt: 0.005
+collision_pair: RARM_JOINT2:LARM_JOINT2 RARM_JOINT3:LARM_JOINT3 RARM_JOINT4:LARM_JOINT4 RARM_JOINT5:LARM_JOINT5
+end_effectors: rarm,RARM_JOINT5,CHEST_JOINT0,0.0,0.0,0.0,-0.57735027,-0.57735027,-0.57735027,-2.0943951023931953,larm,LARM_JOINT5,CHEST_JOINT0,0.0,0.0,0.0,-0.57735027,-0.57735027,-0.57735027,-2.0943951023931953,
+
+# RTC disable/enable settings which will be loaded via RobotHardware
+CollisionDetector.enable: YES
+ServoController.enable: NO
+DataLogger.enable: NOO  # Test case when typo exists
+

--- a/hironx_ros_bridge/src/hironx_ros_bridge/hironx_client.py
+++ b/hironx_ros_bridge/src/hironx_ros_bridge/hironx_client.py
@@ -32,6 +32,7 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import copy
 import math
 import numpy
 import os
@@ -448,22 +449,46 @@ class HIRONX(HrpsysConfigurator2):
             ['fk', "ForwardKinematics"],
             ['ic', "ImpedanceController"],
             ['el', "SoftErrorLimiter"],
+            ['co', "CollisionDetector"],
             ['sc', "ServoController"],
             ['log', "DataLogger"],
             ]
+
+        # Want to move the following to upstream:
+        ## If "(RTC name).enable: NO" is set in RobotHardware conf, remove that RTC
+        ## e.g. "CollisionDetector.enable: NO" in /opt/jsk/etc/HIRONX/hrprtc/Robot.conf
+        if self.ms and self.ms.ref and len(self.ms.ref.get_component_profiles()) > 0:
+            for rtc in copy.deepcopy(rtclist):
+                try:
+                    enable = next(p for p
+                                  in self.ms.ref.get_component_profiles()[0].properties
+                                  if p.name == (rtc[1] + '.enable')).value.value()
+                except StopIteration:
+                    enable = 'YES'
+                if enable == 'NO':
+                    rtclist.remove(rtc)
+                elif enable != 'YES':
+                    print(self.configurator_name +
+                          '\033[31mConfig "' + (rtc[1] + '.enable') + '" is ' +
+                          enable + '. Set YES or NO\033[0m')
+
+        # Specific code to current HIRONX status:
+        ## CollisionDetector.enable is not set in normal conf, but we must remove CollisionDetector
         if self.ms and self.ms.ref and len(self.ms.ref.get_component_profiles()) > 0:
             try:
-                is_co_enabled = next(p for p in self.ms.ref.get_component_profiles()[0].properties
-                                     if p.name == 'is_collision_detector_enabled').value.value()
+                enable = next(p for p
+                              in self.ms.ref.get_component_profiles()[0].properties
+                              if p.name == 'CollisionDetector.enable').value.value()
             except StopIteration:
-                is_co_enabled = 'NO'
-            if is_co_enabled == 'YES':
-                tmpidx = rtclist.index(['el', "SoftErrorLimiter"])
-                rtclist = rtclist[0:tmpidx+1] + [['co', "CollisionDetector"]] + rtclist[tmpidx+1:]
-            elif is_co_enabled != 'NO':
-                print(self.configurator_name +
-                      "\033[31mConfig 'is_collision_detector_enabled' is " + is_co_enabled +
-                      ". Set YES or NO\033[0m")
+                enable = 'NO'
+            if enable == 'NO':
+                try:
+                    rtclist.remove(['co', "CollisionDetector"])
+                except ValueError:  # list.remove(x): x not in list
+                    pass
+        else:
+            rtclist.remove(['co', "CollisionDetector"])
+
         if hasattr(self, 'rmfo'):
             self.ms.load("RemoveForceSensorLinkOffset")
             self.ms.load("AbsoluteForceSensor")

--- a/hironx_ros_bridge/test/test-hironx-rtc-on-off.test
+++ b/hironx_ros_bridge/test/test-hironx-rtc-on-off.test
@@ -1,0 +1,18 @@
+<launch>
+  <arg name="corbaport" default="2809" />
+  <arg name="GUI" default="false" />
+
+  <!-- See https://github.com/start-jsk/rtmros_common/issues/416#issuecomment-46846623 -->
+  <node name="start_omninames" pkg="rtmbuild" type="start_omninames.sh" args="$(arg corbaport)" />
+
+  <include file="$(find hironx_ros_bridge)/launch/hironx_startup.launch" >
+    <arg name="GUI" value="$(arg GUI)" />
+    <arg name="LAUNCH_HRPSYSPY" value="false" />
+    <arg name="REALTIME" value="false" />
+    <arg name="CONF_FILE" value="$(find hironx_ros_bridge)/conf/kawada-hironx_rtc_on_off_test.conf" />
+    <arg name="corbaport" value="$(arg corbaport)" />
+  </include>
+
+  <test type="test_hironx_rtc_on_off.py" pkg="hironx_ros_bridge" test-name="test_hironx_rtc_on_off" time-limit="200" retry="4"
+        args="-ORBInitRef NameService=corbaloc:iiop:localhost:2809/NameService" />
+</launch>

--- a/hironx_ros_bridge/test/test_hironx_fullbody.py
+++ b/hironx_ros_bridge/test/test_hironx_fullbody.py
@@ -5,6 +5,9 @@ from test_hironx import *
 
 class TestHiroFullbody(TestHiro):
 
+    def test_collision_detector_disabled(self):
+        self.assertFalse(['co', "CollisionDetector"] in self.robot.getRTCList())
+
     def fullbody_init (self):
         self.filenames = []
         self.robot.seq_svc.removeJointGroup("larm")

--- a/hironx_ros_bridge/test/test_hironx_rtc_on_off.py
+++ b/hironx_ros_bridge/test/test_hironx_rtc_on_off.py
@@ -1,0 +1,16 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from test_hironx import *
+
+class TestHironxRtcOnOff(TestHiro):
+
+    def test_rtc_on_off(self):
+        self.assertTrue(['co', "CollisionDetector"] in self.robot.getRTCList())
+        self.assertFalse(['sc', "ServoController"] in self.robot.getRTCList())
+        self.assertTrue(['log', "DataLogger"] in self.robot.getRTCList())
+
+
+if __name__ == '__main__':
+    import rostest
+    rostest.rosrun(PKG, 'test_hironx_rtc_on_off', TestHironxRtcOnOff)

